integration_tests/test_statistics.py

```diff
@@ -120,6 +79,36 @@ def test_stdev():
     k = stdev(b)
     assert abs(k - 0.41967844833872525) < eps
 
+def test_mode():
+    a: list[i32]
+    a = [1, 2, 3, 4, 5]
+    i: i32
+    i = mode(a)
+    assert i == 1
+
+    b: list[i32]
+    b = [1, 2, 2, 3, 4, 4, 4, 5]
+    j: i32
+    j = mode(b)
+    assert j == 4
+
+    c: list[i32]
+    c = [1, 2, 2, 3, 4, 4, 5]
+    k: i32
+    k = mode(c)
+    assert k == 2
+
+    d: list[f64]
+    d = [13.2, 24.0, 3.1, 41.9, 5.0]
+    l: f64
+    l = mode(d)
+    assert l == 13.2
+
+    e: list[f64]
+    e = [1.7, 2.1, 2.1, 30.0, 49.7, 49.7, 49.7, 5.0]
+    m: f64
+    m = mode(e)
+    assert m == 49.7
```

Comment:
```suggestion
    assert abs(m - 49.7) < 1e-12
```

@Smit-create Would you mind changing these equality checks over floating points in a similar manner in this whole diff? It would be really helpful.